pydropsonde/circles.py

```diff
@@ -219,33 +219,40 @@ def interpolate_na_sondes(self, method="akima", max_gap=1500, thresh=4):
         return self
 
     @staticmethod
-    def fit2d(x, y, u):
+    def fit2d(x, y, u, weight=1):
+        weight = np.asarray(weight)
         a = np.stack([np.ones_like(x), x, y], axis=-1)
 
         invalid = np.isnan(u) | np.isnan(x) | np.isnan(y)
         # remove values where fewer than 6 sondes are present. Depending on the application, this might be changed.
         under_constraint = np.sum(~invalid, axis=-1) < 6
-        u_cal = np.where(invalid, 0, u)
         a[invalid] = 0
+        sqw = np.where(invalid, 0, np.sqrt(weight))
+        wa = sqw[..., np.newaxis] * a  # np.einsum("...m,...mr->...mr", w, a)
+        u_cal = sqw * np.where(
+            invalid, 0, u
+        )  # np.einsum("...m,...m->...m", w, np.where(invalid, 0, u))
```

Comment:
I'd suggest to remove the comments and pick either implementation. (The comments don't really explain anything not written in the code, and actually seem to do something different)